### DISCUSSION TITLE
AntDTable， table组件，有没有监测table组件已经更新完内容的回调函数，或者Event

### DISCUSSION THREAD

**Discussion by GeordiIGT** [ant-design/ant-design#36125] [Q&A] [answered]
我想对table组件的更新性能进行一下测试，需要进行一下读秒计时，当我使用setData更新dataSource属性的js变量内容以后，我需要等待table组件的内容更新之后再次进行读秒计时，从而得到table组件更新内容的duration，但是我找了很久也没有找到相关的回调函数或者触发的事件，所以想求助大神们有没有知道如何解决的，传统的useEffect监听dataSource变量的方法感觉时间上好像不是特别准确，触发的时机并不是table组件完成更新渲染的时间。求指点

**Comment by huntdream** [comment, marked as answer]
应该可以用React Developer Tools 的 Profiler

**Reply by GeordiIGT** [reply]
> 应该可以用React Developer Tools 的 Profiler

如果想在页面内输出有没有办法呢？React Developer Tools 的功能毕竟还涉及到录制过程，操作起来不是那么的直观，而我这边就是想要直观的表现我的渲染过程经历了多少毫秒，跟其他组件使用相同的数据结构进行性能比较的时候可以一目了然

**Reply by huntdream** [reply]
https://reactjs.org/docs/profiler.html 看看这个

**Reply by GeordiIGT** [reply]
这个好，是我需要的效果！谢谢你的回答

> https://reactjs.org/docs/profiler.html 看看这个
